design/multiple-storage-types-support.md

```diff
@@ -116,171 +116,144 @@ It also provides both resource and privilege isolation for each backend.
+1. All Kinds within a group must exist in the newest version of the Group (unless they were deprecated and have been removed in the latest version)
+1. If a field is added to an older version of the Group, then the field must also be added to the latest version of the Group.
 
-### Versioning
+#### Proposed Custom Resources
 
-Versioning of custom resources defined by Rook is important, and we should carefully consider a design that allows resources to be versioned in a sensible way.
-Think of the following scenario where we have the Ceph and Minio backends both to a level of maturity that we have deemed `stable`.
-Then, if a new storage backend is incorporated into Rook with an early implementation, it should be able to start at `alpha` to properly indicate its nascent level of maturity and stability.
-Therefore, there is a need for each backend to be able to be versioned **independently** of other backends.
+With the intent for Rook's resources to fulfill the desirable properties mentioned above and also meet the Kubernetes versioning guidelines, we propose the following API groups:
 
-Independent versioning is possible by making each storage backend a top level custom resource.
-The same can not be said for an approach of a single cluster CRD that has specific backends as sub-fields.
-To demonstrate this point, let's explore further the scenario above where we have a stable `cluster` CRD that supports `ceph` and `minio`.
-This stable CRD has dropped some properties and gained some properties during its maturity from `alpha` to `stable`.
-To add the support for the brand new storage backend `foostore`, it should be added to the `alpha` CRD because that properly represents its maturity.
-Therefore, we would have the following yaml:
+* `rook.io`: common abstractions that have use across multiple storage backends and types
+* `cluster.rook.io`: common types for clustered/distributed storage, such as `StorageCluster`, `Pool`, `Filesystem`, etc.
+  * `ceph.cluster.rook.io`: Ceph specific `*Spec` types that can be plugged into the common Cluster types.  This Ceph specific API group allows Ceph types to be versioned independently.
+  * `minio.cluster.rook.io`: Similar, but for Minio.
+  * `nexenta.cluster.rook.io`: Similar, but for Nexenta.
+
+This approach to resource definitions and versioning allows Rook to create an experience in `yaml` that would look like the following, where `StorageCluster` is the common/consistent top level resource the user creates, and backend specific configuration can be added as subsections.
+
+`cluster.yaml`:
 
 ```yaml
-apiVersion: rook.io/v1
-kind: StorageCluster
-spec:
-  someStableField: "stableValue"
-  ceph:
-  mino:
----
-apiVersion: rook.io/v1alpha1
+apiVersion: cluster.rook.io/v1
 kind: StorageCluster
 spec:
-  someAlphaField: "alphaValue"
+  version: "myrepo/myimage:v0.1.0"
+  hostNetwork: true
+  placement:
```

Comment:
`placement` and `resources` elements seem like they would belong to components of the backends. Or at least the sub-elements of these would belong to specific backends so i guess it's generic enough to put at the root